lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -34,11 +34,62 @@ static int getTensorRank(Value tensor) {
   return tensorRank;
 }
 
+// Helper function to compute the return type of the reduction function along
+// multiple dimensions. `dimList` specifies a list of dimensions to reduce and
+// `keepDim` preserves the rank of the input tensor.
+static LogicalResult
+computeReductionAlongMultipleDimsType(PatternRewriter &rewriter, Operation *op,
+                                      Value input, Value dimList, bool keepDim,
+                                      BaseTensorType &resultType) {
+  BaseTensorType inputType = input.getType().cast<BaseTensorType>();
+  if (!inputType.hasSizes())
+    return rewriter.notifyMatchFailure(op, "input must be a ranked tensor");
+
+  SmallVector<int64_t> sizes;
+  ArrayRef<int64_t> inputShape = inputType.getSizes();
+  int64_t inputRank = inputShape.size();
+  SmallVector<int64_t> dimIntList;
+  if (matchPattern(dimList, m_TorchConstantIntList(dimIntList))) {
+    for (int64_t dim : dimIntList) {
+      dim = toPositiveDim(dim, inputRank);
+      if (!isValidDim(dim, inputRank))
+        return rewriter.notifyMatchFailure(op, "dim is not a valid dimension");
+    }
+    // `dimSet` contains a list of dimensions to be reduced.
+    DenseSet<int64_t> dimSet(dimIntList.begin(), dimIntList.end());
+    for (auto en : llvm::enumerate(inputShape)) {
+      if (dimSet.contains(en.index())) {
+        // The dimension to be reduced is set to 1 when `keepDim` is true else
+        // it is removed.
+        if (keepDim)
+          sizes.push_back(1);
+      } else {
+        sizes.push_back(en.value());
+      }
+    }
+  } else if (auto listConstruct =
+                 dimList.getDefiningOp<PrimListConstructOp>()) {
+    auto listElements = listConstruct.elements();
+    sizes.resize(keepDim ? inputRank : inputRank - listElements.size(),
+                 ShapedType::kDynamicSize);
+  } else {
+    return rewriter.notifyMatchFailure(
+        op, "unimplemented: dimList is not constructed from list construct");
+  }
+
+  resultType =
+      inputType
+          .getWithSizesAndDtype(llvm::makeArrayRef(sizes), inputType.getDtype())
+          .cast<BaseTensorType>();
+  return success();
+}
+
 // Helper function to compute the return type of the reduction function.
 // `dim` specifies the dimension to reduce and `keepDim` preserves the rank of
 // the input tensor.
-static Type computeReductionType(PatternRewriter &rewriter, Operation *op,
-                                 Value input, Value dim, bool keepDim) {
+static Type computeReductionAlongDimType(PatternRewriter &rewriter,
```

Comment:
This is just a special case of the `computeReductionAlongMultipleDimsType`, so you should rewrite it using that function.